chart_utils.py

```diff
@@ -0,0 +1,244 @@
+    """
+    Load loan data and main chart data for the specified protocol.
+    :param protocol_name: Protocol name.
+    :return: DataFrames containing loan data and main chart data.
+    """
+    return src.helpers.load_data(protocol=protocol_name)
+
+
+def get_protocol_data_mappings(
+    current_pair: str, stable_coin_pair: str, protocols: list[str]
+) -> Tuple[Dict[str, dict], Dict[str, dict]]:
+    """
+    Get protocol data mappings for main chart data and loans data.
+
+    :param current_pair: The current pair for which data is to be fetched.
+    :param stable_coin_pair: The stable coin pair to check against.
+    :param protocols: List of protocols for which data is to be fetched.
+    :return: tuple of dictionaries containing:
+        - protocol_main_chart_data: Mapping of protocol names to their main chart data.
+        - protocol_loans_data: Mapping of protocol names to their loans data.
+    """
+
+    protocol_main_chart_data: Dict[str, dict] = {}
+    protocol_loans_data: Dict[str, dict] = {}
+
+    for protocol_name in protocols:
+        main_chart_data, loans_data = get_load_data(protocol_name)
+        protocol_loans_data[protocol_name] = loans_data
+
+        if current_pair == stable_coin_pair:
+            protocol_main_chart_data[protocol_name] = create_stablecoin_bundle(
+                main_chart_data
+            )[current_pair]
+        else:
+            protocol_main_chart_data[protocol_name] = main_chart_data[current_pair]
+
+    return protocol_main_chart_data, protocol_loans_data
+
+
+def load_stats_data() -> (
+    Tuple[pd.DataFrame, pd.DataFrame, pd.DataFrame, pd.DataFrame, pd.DataFrame]
+):
+    """
+    Load general stats, supply stats, collateral stats, and debt stats data.
+    :return: tuple of DataFrames containing general stats, supply stats, collateral stats, debt stats, and utilization stats.
+    """
+    BASE_GS_PATH = f"gs://{src.helpers.GS_BUCKET_NAME}/data"
+    read_parquet_with_protocol_index = partial(pd.read_parquet, engine="fastparquet")
+
+    @streamlit.cache_data(ttl=300)
+    def read_and_set_index(file_path: str) -> pd.DataFrame:
+        """
+        Read a parquet file and set the index to 'Protocol'.
+        :param file_path: file_path
+        :return: DataFrame
+        """
+        return read_parquet_with_protocol_index(file_path).set_index("Protocol")
+
+    # Read the parquet files
+    general_stats = read_and_set_index(f"{BASE_GS_PATH}/general_stats.parquet")
+    supply_stats = read_and_set_index(f"{BASE_GS_PATH}/supply_stats.parquet")
+    collateral_stats = read_and_set_index(f"{BASE_GS_PATH}/collateral_stats.parquet")
+    debt_stats = read_and_set_index(f"{BASE_GS_PATH}/debt_stats.parquet")
+
+    # Calculate TVL (USD)
+    general_stats["TVL (USD)"] = (
+        supply_stats["Total supply (USD)"] - general_stats["Total debt (USD)"]
+    )
+    utilization_stats = read_and_set_index(f"{BASE_GS_PATH}/utilization_stats.parquet")
```

Comment:
Move next to where the other files are loaded for the sake of consistency.

Reply:
I have all these kind of logic in one file, chart_utils.py